src/auth0-sso-login.js

```diff
@@ -122,87 +134,39 @@ export default class auth {
    * @return {Promise<>} empty resolved promise after successful login; rejected promise with error
    *                     otherwise
    */
-  ensureLoggedIn(configuration = { enableLockWidget: true, forceTokenRefresh: false }) {
+  ensureLoggedIn(configuration = { enabledHostedLogin: true, forceTokenRefresh: false }) {
     // if there is still a valid token, there is no need to initiate the login process
-    const latestAuthResult = this.getLatestAuthResult();
+    const latestAuthResult = this.getIdToken();
     if (!configuration.forceTokenRefresh && latestAuthResult &&
       this.tokenExpiryManager.getRemainingMillisToTokenExpiry() > 0) {
       return Promise.resolve();
     }
 
-    let options = {
-      auth: {
-        params: {
-          responseType: 'id_token token',
-        },
-        redirect: false,
-      },
-      closable: false,
-    };
-    if (this.config.auth0LockOptions) {
-      options = Object.assign(options, this.config.auth0LockOptions);
-    }
+    this.redirectHandler.attemptRedirect();
 
     const authPromise = this.renewAuthSequencePromise
       .then(() => this.renewAuth())
       .catch((e) => {
-        // if auth0 lock is not enabled, error out
-        if (!configuration.enableLockWidget) {
+        // if universal login is not enabled, error out
+        if (!configuration.enabledHostedLogin) {
           return Promise.reject(e);
         }
-        this.log({ title: 'Renew authorization did not succeed, falling back to login widget', error: e });
-        return new Promise((resolve, reject) => {
-          const lock = auth0LockFactory.createAuth0Lock(this.config.clientId, this.config.domain,
-            options);
-          lock.on('authenticated', (authResult) => {
-            this.renewAuth()
-              .then(() => {
-                lock.getUserInfo(authResult.accessToken, (error, profile) => {
-                  lock.hide();
-                  if (error) {
-                    this.log({ title: 'Error while retrieving user information after successful Auth0Lock authentication', error });
-                    resolve({ idToken: authResult.idToken, sub: null });
-                  } else {
-                    resolve({
-                      idToken: authResult.idToken,
-                      sub: profile.sub,
-                    });
-                  }
-                });
-              })
-              .catch((error) => {
-                this.log({ title: 'Error while calling renewAuth after successful Auth0Lock authentication', error });
-                resolve({ idToken: authResult.idToken, sub: null });
-              });
-          });
-
-          lock.on('authorization_error', (error) => {
-            this.log({ title: 'authorization error', error });
-            reject(error);
-          });
 
-          lock.show();
-        });
+        this.logger.log({ title: 'Renew authorization did not succeed, falling back to login widget', error: e });
```

Comment:
I'd be more explicit in the log message: `... falling back to Auth0 redirect` or something similar.

Reply:
Fixed.